code/config/Config.rst
                                      1 ;--------------------------------------------------------
                                      2 ; File Created by SDCC : free open source ISO C Compiler
                                      3 ; Version 4.5.0 #15242 (Mac OS X ppc)
                                      4 ;--------------------------------------------------------
                                      5 	.module Config
                                      6 	
                                      7 	.optsdcc -mmcs51 --model-small
                                      8 ;--------------------------------------------------------
                                      9 ; Public variables in this module
                                     10 ;--------------------------------------------------------
                                     11 	.globl _CONF_RADIO_PUT_PARM_2
                                     12 	.globl _CONF_READ_RAIDO_FREQ
                                     13 	.globl _IapProgramByte
                                     14 	.globl _IapReadByte
                                     15 	.globl _IapEraseSector
                                     16 	.globl _CCF0
                                     17 	.globl _CCF1
                                     18 	.globl _CCF2
                                     19 	.globl _CR
                                     20 	.globl _CF
                                     21 	.globl _RI
                                     22 	.globl _TI
                                     23 	.globl _RB8
                                     24 	.globl _TB8
                                     25 	.globl _REN
                                     26 	.globl _SM2
                                     27 	.globl _SM1
                                     28 	.globl _SM0
                                     29 	.globl _IT0
                                     30 	.globl _IE0
                                     31 	.globl _IT1
                                     32 	.globl _IE1
                                     33 	.globl _TR0
                                     34 	.globl _TF0
                                     35 	.globl _TR1
                                     36 	.globl _TF1
                                     37 	.globl _PX0
                                     38 	.globl _PT0
                                     39 	.globl _PX1
                                     40 	.globl _PT1
                                     41 	.globl _PS
                                     42 	.globl _PADC
                                     43 	.globl _PLVD
                                     44 	.globl _PPCA
                                     45 	.globl _EX0
                                     46 	.globl _ET0
                                     47 	.globl _EX1
                                     48 	.globl _ET1
                                     49 	.globl _ES
                                     50 	.globl _EADC
                                     51 	.globl _ELVD
                                     52 	.globl _EA
                                     53 	.globl _P77
                                     54 	.globl _P76
                                     55 	.globl _P75
                                     56 	.globl _P74
                                     57 	.globl _P73
                                     58 	.globl _P72
                                     59 	.globl _P71
                                     60 	.globl _P70
                                     61 	.globl _P67
                                     62 	.globl _P66
                                     63 	.globl _P65
                                     64 	.globl _P64
                                     65 	.globl _P63
                                     66 	.globl _P62
                                     67 	.globl _P61
                                     68 	.globl _P60
                                     69 	.globl _P57
                                     70 	.globl _P56
                                     71 	.globl _P55
                                     72 	.globl _P54
                                     73 	.globl _P53
                                     74 	.globl _P52
                                     75 	.globl _P51
                                     76 	.globl _P50
                                     77 	.globl _P47
                                     78 	.globl _P46
                                     79 	.globl _P45
                                     80 	.globl _P44
                                     81 	.globl _P43
                                     82 	.globl _P42
                                     83 	.globl _P41
                                     84 	.globl _P40
                                     85 	.globl _P37
                                     86 	.globl _P36
                                     87 	.globl _P35
                                     88 	.globl _P34
                                     89 	.globl _P33
                                     90 	.globl _P32
                                     91 	.globl _P31
                                     92 	.globl _P30
                                     93 	.globl _P27
                                     94 	.globl _P26
                                     95 	.globl _P25
                                     96 	.globl _P24
                                     97 	.globl _P23
                                     98 	.globl _P22
                                     99 	.globl _P21
                                    100 	.globl _P20
                                    101 	.globl _P17
                                    102 	.globl _P16
                                    103 	.globl _P15
                                    104 	.globl _P14
                                    105 	.globl _P13
                                    106 	.globl _P12
                                    107 	.globl _P11
                                    108 	.globl _P10
                                    109 	.globl _P07
                                    110 	.globl _P06
                                    111 	.globl _P05
                                    112 	.globl _P04
                                    113 	.globl _P03
                                    114 	.globl _P02
                                    115 	.globl _P01
                                    116 	.globl _P00
                                    117 	.globl _P
                                    118 	.globl _F1
                                    119 	.globl _OV
                                    120 	.globl _RS0
                                    121 	.globl _RS1
                                    122 	.globl _F0
                                    123 	.globl _AC
                                    124 	.globl _CY
                                    125 	.globl _PWMFDCR
                                    126 	.globl _PWMIF
                                    127 	.globl _PWMCR
                                    128 	.globl _PWMCFG
                                    129 	.globl _CMPCR2
                                    130 	.globl _CMPCR1
                                    131 	.globl _CCAP2H
                                    132 	.globl _CCAP1H
                                    133 	.globl _CCAP0H
                                    134 	.globl _PCA_PWM2
                                    135 	.globl _PCA_PWM1
                                    136 	.globl _PCA_PWM0
                                    137 	.globl _CCAP2L
                                    138 	.globl _CCAP1L
                                    139 	.globl _CCAP0L
                                    140 	.globl _CCAPM2
                                    141 	.globl _CCAPM1
                                    142 	.globl _CCAPM0
                                    143 	.globl _CH
                                    144 	.globl _CL
                                    145 	.globl _CMOD
                                    146 	.globl _CCON
                                    147 	.globl _IAP_CONTR
                                    148 	.globl _IAP_TRIG
                                    149 	.globl _IAP_CMD
                                    150 	.globl _IAP_ADDRL
                                    151 	.globl _IAP_ADDRH
                                    152 	.globl _IAP_DATA
                                    153 	.globl _SPDAT
                                    154 	.globl _SPCTL
                                    155 	.globl _SPSTAT
                                    156 	.globl _ADC_RESL
                                    157 	.globl _ADC_RES
                                    158 	.globl _ADC_CONTR
                                    159 	.globl _SADEN
                                    160 	.globl _SADDR
                                    161 	.globl _S4BUF
                                    162 	.globl _S4CON
                                    163 	.globl _S3BUF
                                    164 	.globl _S3CON
                                    165 	.globl _S2BUF
                                    166 	.globl _S2CON
                                    167 	.globl _SBUF
                                    168 	.globl _SCON
                                    169 	.globl _WDT_CONTR
                                    170 	.globl _WKTCH
                                    171 	.globl _WKTCL
                                    172 	.globl _T2L
                                    173 	.globl _T2H
                                    174 	.globl _T3L
                                    175 	.globl _T3H
                                    176 	.globl _T4L
                                    177 	.globl _T4H
                                    178 	.globl _T3T4M
                                    179 	.globl _T4T3M
                                    180 	.globl _TH1
                                    181 	.globl _TH0
                                    182 	.globl _TL1
                                    183 	.globl _TL0
                                    184 	.globl _TMOD
                                    185 	.globl _TCON
                                    186 	.globl _INT_CLKO
                                    187 	.globl _IP2
                                    188 	.globl _IE2
                                    189 	.globl _IP
                                    190 	.globl _IE
                                    191 	.globl _P_SW2
                                    192 	.globl _P1ASF
                                    193 	.globl _BUS_SPEED
                                    194 	.globl _CLK_DIV
                                    195 	.globl _P_SW1
                                    196 	.globl _AUXR1
                                    197 	.globl _AUXR
                                    198 	.globl _PCON
                                    199 	.globl _P7M1
                                    200 	.globl _P7M0
                                    201 	.globl _P6M1
                                    202 	.globl _P6M0
                                    203 	.globl _P5M1
                                    204 	.globl _P5M0
                                    205 	.globl _P4M1
                                    206 	.globl _P4M0
                                    207 	.globl _P3M1
                                    208 	.globl _P3M0
                                    209 	.globl _P2M1
                                    210 	.globl _P2M0
                                    211 	.globl _P1M1
                                    212 	.globl _P1M0
                                    213 	.globl _P0M1
                                    214 	.globl _P0M0
                                    215 	.globl _P7
                                    216 	.globl _P6
                                    217 	.globl _P5
                                    218 	.globl _P4
                                    219 	.globl _P3
                                    220 	.globl _P2
                                    221 	.globl _P1
                                    222 	.globl _P0
                                    223 	.globl _DPH
                                    224 	.globl _DPL
                                    225 	.globl _SP
                                    226 	.globl _PSW
                                    227 	.globl _B
                                    228 	.globl _ACC
                                    229 	.globl _sys_radio_index_max
                                    230 	.globl _sys_radio_index
                                    231 	.globl _sys_freq
                                    232 	.globl _sys_sleep_mode
                                    233 	.globl _sys_vol
                                    234 	.globl _CONF_SET_VOL
                                    235 	.globl _CONF_SET_FREQ
                                    236 	.globl _CONF_CHANGE_SLEEP_MODE
                                    237 	.globl _CONF_RADIO_ERASE
                                    238 	.globl _CONF_RADIO_PUT
                                    239 	.globl _CONF_SET_INDEX_MAX
                                    240 	.globl _CONF_SYS_INIT
                                    241 	.globl _CONF_GET_RADIO_INDEX
                                    242 ;--------------------------------------------------------
                                    243 ; special function registers
                                    244 ;--------------------------------------------------------
                                    245 	.area RSEG    (ABS,DATA)
      000000                        246 	.org 0x0000
                           0000E0   247 _ACC	=	0x00e0
                           0000F0   248 _B	=	0x00f0
                           0000D0   249 _PSW	=	0x00d0
                           000081   250 _SP	=	0x0081
                           000082   251 _DPL	=	0x0082
                           000083   252 _DPH	=	0x0083
                           000080   253 _P0	=	0x0080
                           000090   254 _P1	=	0x0090
                           0000A0   255 _P2	=	0x00a0
                           0000B0   256 _P3	=	0x00b0
                           0000C0   257 _P4	=	0x00c0
                           0000C8   258 _P5	=	0x00c8
                           0000E8   259 _P6	=	0x00e8
                           0000F8   260 _P7	=	0x00f8
                           000094   261 _P0M0	=	0x0094
                           000093   262 _P0M1	=	0x0093
                           000092   263 _P1M0	=	0x0092
                           000091   264 _P1M1	=	0x0091
                           000096   265 _P2M0	=	0x0096
                           000095   266 _P2M1	=	0x0095
                           0000B2   267 _P3M0	=	0x00b2
                           0000B1   268 _P3M1	=	0x00b1
                           0000B4   269 _P4M0	=	0x00b4
                           0000B3   270 _P4M1	=	0x00b3
                           0000CA   271 _P5M0	=	0x00ca
                           0000C9   272 _P5M1	=	0x00c9
                           0000CC   273 _P6M0	=	0x00cc
                           0000CB   274 _P6M1	=	0x00cb
                           0000E2   275 _P7M0	=	0x00e2
                           0000E1   276 _P7M1	=	0x00e1
                           000087   277 _PCON	=	0x0087
                           00008E   278 _AUXR	=	0x008e
                           0000A2   279 _AUXR1	=	0x00a2
                           0000A2   280 _P_SW1	=	0x00a2
                           000097   281 _CLK_DIV	=	0x0097
                           0000A1   282 _BUS_SPEED	=	0x00a1
                           00009D   283 _P1ASF	=	0x009d
                           0000BA   284 _P_SW2	=	0x00ba
                           0000A8   285 _IE	=	0x00a8
                           0000B8   286 _IP	=	0x00b8
                           0000AF   287 _IE2	=	0x00af
                           0000B5   288 _IP2	=	0x00b5
                           00008F   289 _INT_CLKO	=	0x008f
                           000088   290 _TCON	=	0x0088
                           000089   291 _TMOD	=	0x0089
                           00008A   292 _TL0	=	0x008a
                           00008B   293 _TL1	=	0x008b
                           00008C   294 _TH0	=	0x008c
                           00008D   295 _TH1	=	0x008d
                           0000D1   296 _T4T3M	=	0x00d1
                           0000D1   297 _T3T4M	=	0x00d1
                           0000D2   298 _T4H	=	0x00d2
                           0000D3   299 _T4L	=	0x00d3
                           0000D4   300 _T3H	=	0x00d4
                           0000D5   301 _T3L	=	0x00d5
                           0000D6   302 _T2H	=	0x00d6
                           0000D7   303 _T2L	=	0x00d7
                           0000AA   304 _WKTCL	=	0x00aa
                           0000AB   305 _WKTCH	=	0x00ab
                           0000C1   306 _WDT_CONTR	=	0x00c1
                           000098   307 _SCON	=	0x0098
                           000099   308 _SBUF	=	0x0099
                           00009A   309 _S2CON	=	0x009a
                           00009B   310 _S2BUF	=	0x009b
                           0000AC   311 _S3CON	=	0x00ac
                           0000AD   312 _S3BUF	=	0x00ad
                           000084   313 _S4CON	=	0x0084
                           000085   314 _S4BUF	=	0x0085
                           0000A9   315 _SADDR	=	0x00a9
                           0000B9   316 _SADEN	=	0x00b9
                           0000BC   317 _ADC_CONTR	=	0x00bc
                           0000BD   318 _ADC_RES	=	0x00bd
                           0000BE   319 _ADC_RESL	=	0x00be
                           0000CD   320 _SPSTAT	=	0x00cd
                           0000CE   321 _SPCTL	=	0x00ce
                           0000CF   322 _SPDAT	=	0x00cf
                           0000C2   323 _IAP_DATA	=	0x00c2
                           0000C3   324 _IAP_ADDRH	=	0x00c3
                           0000C4   325 _IAP_ADDRL	=	0x00c4
                           0000C5   326 _IAP_CMD	=	0x00c5

code/config/Config.asm
;--------------------------------------------------------
; File Created by SDCC : free open source ISO C Compiler
; Version 4.5.0 #15242 (Mac OS X ppc)
;--------------------------------------------------------
	.module Config
	
	.optsdcc -mmcs51 --model-small
;--------------------------------------------------------
; Public variables in this module
;--------------------------------------------------------
	.globl _CONF_RADIO_PUT_PARM_2
	.globl _CONF_READ_RAIDO_FREQ
	.globl _IapProgramByte
	.globl _IapReadByte
	.globl _IapEraseSector
	.globl _CCF0
	.globl _CCF1
	.globl _CCF2
	.globl _CR
	.globl _CF
	.globl _RI
	.globl _TI
	.globl _RB8
	.globl _TB8
	.globl _REN
	.globl _SM2
	.globl _SM1
	.globl _SM0
	.globl _IT0
	.globl _IE0
	.globl _IT1
	.globl _IE1
	.globl _TR0
	.globl _TF0
	.globl _TR1
	.globl _TF1
	.globl _PX0
	.globl _PT0
	.globl _PX1
	.globl _PT1
	.globl _PS
	.globl _PADC
	.globl _PLVD
	.globl _PPCA
	.globl _EX0
	.globl _ET0
	.globl _EX1
	.globl _ET1
	.globl _ES
	.globl _EADC
	.globl _ELVD
	.globl _EA
	.globl _P77
	.globl _P76
	.globl _P75
	.globl _P74
	.globl _P73
	.globl _P72
	.globl _P71
	.globl _P70
	.globl _P67
	.globl _P66
	.globl _P65
	.globl _P64
	.globl _P63
	.globl _P62
	.globl _P61
	.globl _P60
	.globl _P57
	.globl _P56
	.globl _P55
	.globl _P54
	.globl _P53
	.globl _P52
	.globl _P51
	.globl _P50
	.globl _P47
	.globl _P46
	.globl _P45
	.globl _P44
	.globl _P43
	.globl _P42
	.globl _P41
	.globl _P40
	.globl _P37
	.globl _P36
	.globl _P35
	.globl _P34
	.globl _P33
	.globl _P32
	.globl _P31
	.globl _P30
	.globl _P27
	.globl _P26
	.globl _P25
	.globl _P24
	.globl _P23
	.globl _P22
	.globl _P21
	.globl _P20
	.globl _P17
	.globl _P16
	.globl _P15
	.globl _P14
	.globl _P13
	.globl _P12
	.globl _P11
	.globl _P10
	.globl _P07
	.globl _P06
	.globl _P05
	.globl _P04
	.globl _P03
	.globl _P02
	.globl _P01
	.globl _P00
	.globl _P
	.globl _F1
	.globl _OV
	.globl _RS0
	.globl _RS1
	.globl _F0
	.globl _AC
	.globl _CY
	.globl _PWMFDCR
	.globl _PWMIF
	.globl _PWMCR
	.globl _PWMCFG
	.globl _CMPCR2
	.globl _CMPCR1
	.globl _CCAP2H
	.globl _CCAP1H
	.globl _CCAP0H
	.globl _PCA_PWM2
	.globl _PCA_PWM1
	.globl _PCA_PWM0
	.globl _CCAP2L
	.globl _CCAP1L
	.globl _CCAP0L
	.globl _CCAPM2
	.globl _CCAPM1
	.globl _CCAPM0
	.globl _CH
	.globl _CL
	.globl _CMOD
	.globl _CCON
	.globl _IAP_CONTR
	.globl _IAP_TRIG
	.globl _IAP_CMD
	.globl _IAP_ADDRL
	.globl _IAP_ADDRH
	.globl _IAP_DATA
	.globl _SPDAT
	.globl _SPCTL
	.globl _SPSTAT
	.globl _ADC_RESL
	.globl _ADC_RES
	.globl _ADC_CONTR
	.globl _SADEN
	.globl _SADDR
	.globl _S4BUF
	.globl _S4CON
	.globl _S3BUF
	.globl _S3CON
	.globl _S2BUF
	.globl _S2CON
	.globl _SBUF
	.globl _SCON
	.globl _WDT_CONTR
	.globl _WKTCH
	.globl _WKTCL
	.globl _T2L
	.globl _T2H
	.globl _T3L
	.globl _T3H
	.globl _T4L
	.globl _T4H
	.globl _T3T4M
	.globl _T4T3M
	.globl _TH1
	.globl _TH0
	.globl _TL1
	.globl _TL0
	.globl _TMOD
	.globl _TCON
	.globl _INT_CLKO
	.globl _IP2
	.globl _IE2
	.globl _IP
	.globl _IE
	.globl _P_SW2
	.globl _P1ASF
	.globl _BUS_SPEED
	.globl _CLK_DIV
	.globl _P_SW1
	.globl _AUXR1
	.globl _AUXR
	.globl _PCON
	.globl _P7M1
	.globl _P7M0
	.globl _P6M1
	.globl _P6M0
	.globl _P5M1
	.globl _P5M0
	.globl _P4M1
	.globl _P4M0
	.globl _P3M1
	.globl _P3M0
	.globl _P2M1
	.globl _P2M0
	.globl _P1M1
	.globl _P1M0
	.globl _P0M1
	.globl _P0M0
	.globl _P7
	.globl _P6
	.globl _P5
	.globl _P4
	.globl _P3
	.globl _P2
	.globl _P1
	.globl _P0
	.globl _DPH
	.globl _DPL
	.globl _SP
	.globl _PSW
	.globl _B
	.globl _ACC
	.globl _sys_radio_index_max
	.globl _sys_radio_index
	.globl _sys_freq
	.globl _sys_sleep_mode
	.globl _sys_vol
	.globl _CONF_SET_VOL
	.globl _CONF_SET_FREQ
	.globl _CONF_CHANGE_SLEEP_MODE
	.globl _CONF_RADIO_ERASE
	.globl _CONF_RADIO_PUT
	.globl _CONF_SET_INDEX_MAX
	.globl _CONF_SYS_INIT
	.globl _CONF_GET_RADIO_INDEX
;--------------------------------------------------------
; special function registers
;--------------------------------------------------------
	.area RSEG    (ABS,DATA)
	.org 0x0000
_ACC	=	0x00e0
_B	=	0x00f0
_PSW	=	0x00d0
_SP	=	0x0081
_DPL	=	0x0082
_DPH	=	0x0083
_P0	=	0x0080
_P1	=	0x0090
_P2	=	0x00a0
_P3	=	0x00b0
_P4	=	0x00c0
_P5	=	0x00c8
_P6	=	0x00e8
_P7	=	0x00f8
_P0M0	=	0x0094
_P0M1	=	0x0093
_P1M0	=	0x0092
_P1M1	=	0x0091
_P2M0	=	0x0096
_P2M1	=	0x0095
_P3M0	=	0x00b2
_P3M1	=	0x00b1
_P4M0	=	0x00b4
_P4M1	=	0x00b3
_P5M0	=	0x00ca
_P5M1	=	0x00c9
_P6M0	=	0x00cc
_P6M1	=	0x00cb
_P7M0	=	0x00e2
_P7M1	=	0x00e1
_PCON	=	0x0087
_AUXR	=	0x008e
_AUXR1	=	0x00a2
_P_SW1	=	0x00a2
_CLK_DIV	=	0x0097
_BUS_SPEED	=	0x00a1
_P1ASF	=	0x009d
_P_SW2	=	0x00ba
_IE	=	0x00a8
_IP	=	0x00b8
_IE2	=	0x00af
_IP2	=	0x00b5
_INT_CLKO	=	0x008f
_TCON	=	0x0088
_TMOD	=	0x0089
_TL0	=	0x008a
_TL1	=	0x008b
_TH0	=	0x008c
_TH1	=	0x008d
_T4T3M	=	0x00d1
_T3T4M	=	0x00d1
_T4H	=	0x00d2
_T4L	=	0x00d3
_T3H	=	0x00d4
_T3L	=	0x00d5
_T2H	=	0x00d6
_T2L	=	0x00d7
_WKTCL	=	0x00aa
_WKTCH	=	0x00ab
_WDT_CONTR	=	0x00c1
_SCON	=	0x0098
_SBUF	=	0x0099
_S2CON	=	0x009a
_S2BUF	=	0x009b
_S3CON	=	0x00ac
_S3BUF	=	0x00ad
_S4CON	=	0x0084
_S4BUF	=	0x0085
_SADDR	=	0x00a9
_SADEN	=	0x00b9
_ADC_CONTR	=	0x00bc
_ADC_RES	=	0x00bd
_ADC_RESL	=	0x00be
_SPSTAT	=	0x00cd
_SPCTL	=	0x00ce
_SPDAT	=	0x00cf
_IAP_DATA	=	0x00c2
_IAP_ADDRH	=	0x00c3
_IAP_ADDRL	=	0x00c4
_IAP_CMD	=	0x00c5
_IAP_TRIG	=	0x00c6
_IAP_CONTR	=	0x00c7
_CCON	=	0x00d8
_CMOD	=	0x00d9
_CL	=	0x00e9
_CH	=	0x00f9
_CCAPM0	=	0x00da
_CCAPM1	=	0x00db
_CCAPM2	=	0x00dc
_CCAP0L	=	0x00ea
_CCAP1L	=	0x00eb
_CCAP2L	=	0x00ec
_PCA_PWM0	=	0x00f2
_PCA_PWM1	=	0x00f3
_PCA_PWM2	=	0x00f4
_CCAP0H	=	0x00fa
_CCAP1H	=	0x00fb
_CCAP2H	=	0x00fc
_CMPCR1	=	0x00e6
_CMPCR2	=	0x00e7
_PWMCFG	=	0x00f1
_PWMCR	=	0x00f5
_PWMIF	=	0x00f6
_PWMFDCR	=	0x00f7
;--------------------------------------------------------
; special function bits
;--------------------------------------------------------
	.area RSEG    (ABS,DATA)
	.org 0x0000
_CY	=	0x00d7
_AC	=	0x00d6
_F0	=	0x00d5
_RS1	=	0x00d4
_RS0	=	0x00d3
_OV	=	0x00d2
_F1	=	0x00d1
_P	=	0x00d0
_P00	=	0x0080
_P01	=	0x0081
_P02	=	0x0082
_P03	=	0x0083
_P04	=	0x0084
_P05	=	0x0085
_P06	=	0x0086
_P07	=	0x0087
_P10	=	0x0090
_P11	=	0x0091
_P12	=	0x0092
_P13	=	0x0093
_P14	=	0x0094
_P15	=	0x0095
_P16	=	0x0096
_P17	=	0x0097
_P20	=	0x00a0
_P21	=	0x00a1
_P22	=	0x00a2
_P23	=	0x00a3
_P24	=	0x00a4
_P25	=	0x00a5
_P26	=	0x00a6
_P27	=	0x00a7
_P30	=	0x00b0
_P31	=	0x00b1
_P32	=	0x00b2
_P33	=	0x00b3
_P34	=	0x00b4
_P35	=	0x00b5
_P36	=	0x00b6
_P37	=	0x00b7
_P40	=	0x00c0
_P41	=	0x00c1
_P42	=	0x00c2
_P43	=	0x00c3
_P44	=	0x00c4
_P45	=	0x00c5
_P46	=	0x00c6
_P47	=	0x00c7
_P50	=	0x00c8
_P51	=	0x00c9
_P52	=	0x00ca
_P53	=	0x00cb
_P54	=	0x00cc
_P55	=	0x00cd
_P56	=	0x00ce
_P57	=	0x00cf
_P60	=	0x00e8
_P61	=	0x00e9
_P62	=	0x00ea
_P63	=	0x00eb
_P64	=	0x00ec
_P65	=	0x00ed
_P66	=	0x00ee
_P67	=	0x00ef
_P70	=	0x00f8
_P71	=	0x00f9
_P72	=	0x00fa
_P73	=	0x00fb
_P74	=	0x00fc
_P75	=	0x00fd
_P76	=	0x00fe
_P77	=	0x00ff
_EA	=	0x00af
_ELVD	=	0x00ae
_EADC	=	0x00ad
_ES	=	0x00ac
_ET1	=	0x00ab
_EX1	=	0x00aa
_ET0	=	0x00a9
_EX0	=	0x00a8
_PPCA	=	0x00bf
_PLVD	=	0x00be
_PADC	=	0x00bd
_PS	=	0x00bc
_PT1	=	0x00bb
_PX1	=	0x00ba
_PT0	=	0x00b9
_PX0	=	0x00b8
_TF1	=	0x008f
_TR1	=	0x008e
_TF0	=	0x008d
_TR0	=	0x008c
_IE1	=	0x008b
_IT1	=	0x008a
_IE0	=	0x0089
_IT0	=	0x0088
_SM0	=	0x009f
_SM1	=	0x009e
_SM2	=	0x009d
_REN	=	0x009c
_TB8	=	0x009b
_RB8	=	0x009a
_TI	=	0x0099
_RI	=	0x0098
_CF	=	0x00df
_CR	=	0x00de
_CCF2	=	0x00da
_CCF1	=	0x00d9
_CCF0	=	0x00d8
;--------------------------------------------------------
; overlayable register banks
;--------------------------------------------------------
	.area REG_BANK_0	(REL,OVR,DATA)
	.ds 8
;--------------------------------------------------------
; internal ram data
;--------------------------------------------------------
	.area DSEG    (DATA)
_sys_vol::
	.ds 1
_sys_sleep_mode::
	.ds 1
_sys_freq::
	.ds 2
_sys_radio_index::
	.ds 1
_sys_radio_index_max::
	.ds 1
;--------------------------------------------------------
; overlayable items in internal ram
;--------------------------------------------------------
	.area	OSEG    (OVR,DATA)
	.area	OSEG    (OVR,DATA)
	.area	OSEG    (OVR,DATA)
	.area	OSEG    (OVR,DATA)
_CONF_RADIO_PUT_PARM_2:
	.ds 2
	.area	OSEG    (OVR,DATA)
	.area	OSEG    (OVR,DATA)
;--------------------------------------------------------
; indirectly addressable internal ram data
;--------------------------------------------------------
	.area ISEG    (DATA)
;--------------------------------------------------------
; absolute internal ram data
;--------------------------------------------------------
	.area IABS    (ABS,DATA)
	.area IABS    (ABS,DATA)
;--------------------------------------------------------
; bit data
;--------------------------------------------------------
	.area BSEG    (BIT)
;--------------------------------------------------------
; paged external ram data
;--------------------------------------------------------
	.area PSEG    (PAG,XDATA)
;--------------------------------------------------------
; uninitialized external ram data
;--------------------------------------------------------
	.area XSEG    (XDATA)
;--------------------------------------------------------
; absolute external ram data
;--------------------------------------------------------
	.area XABS    (ABS,XDATA)
;--------------------------------------------------------
; initialized external ram data
;--------------------------------------------------------
	.area XISEG   (XDATA)
	.area HOME    (CODE)
	.area GSINIT0 (CODE)
	.area GSINIT1 (CODE)
	.area GSINIT2 (CODE)
	.area GSINIT3 (CODE)
	.area GSINIT4 (CODE)
	.area GSINIT5 (CODE)
	.area GSINIT  (CODE)
	.area GSFINAL (CODE)
	.area CSEG    (CODE)
;--------------------------------------------------------
; global & static initialisations
;--------------------------------------------------------
	.area HOME    (CODE)
	.area GSINIT  (CODE)
	.area GSFINAL (CODE)
	.area GSINIT  (CODE)
;	code/config/Config.c:7: uint8_t sys_vol = 10;
	mov	_sys_vol,#0x0a
;	code/config/Config.c:9: uint8_t sys_sleep_mode = 0x01;
	mov	_sys_sleep_mode,#0x01
;	code/config/Config.c:10: uint16_t sys_freq = 7640; // 76.4Mhz
	mov	_sys_freq,#0xd8
	mov	(_sys_freq + 1),#0x1d
;	code/config/Config.c:12: uint8_t sys_radio_index = 0x00;
	mov	_sys_radio_index,#0x00
;	code/config/Config.c:13: uint8_t sys_radio_index_max = 0x00;
	mov	_sys_radio_index_max,#0x00
;--------------------------------------------------------
; Home
;--------------------------------------------------------
	.area HOME    (CODE)
	.area HOME    (CODE)
;--------------------------------------------------------
; code
;--------------------------------------------------------
	.area CSEG    (CODE)
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_READ_RAIDO_FREQ'
;------------------------------------------------------------
;addr          Allocated to registers 
;------------------------------------------------------------
;	code/config/Config.c:18: uint16_t CONF_READ_RAIDO_FREQ(uint16_t addr)
;	-----------------------------------------
;	 function CONF_READ_RAIDO_FREQ
;	-----------------------------------------
_CONF_READ_RAIDO_FREQ:
	ar7 = 0x07
	ar6 = 0x06
	ar5 = 0x05
	ar4 = 0x04
	ar3 = 0x03
	ar2 = 0x02
	ar1 = 0x01
	ar0 = 0x00
;	code/config/Config.c:20: return 0;
	mov	dptr,#0x0000
;	code/config/Config.c:21: }
	ret
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_SET_VOL'
;------------------------------------------------------------
;vol           Allocated to registers 
;------------------------------------------------------------
;	code/config/Config.c:23: void CONF_SET_VOL(uint8_t vol)
;	-----------------------------------------
;	 function CONF_SET_VOL
;	-----------------------------------------
_CONF_SET_VOL:
	mov	_sys_vol,dpl
;	code/config/Config.c:25: sys_vol = vol;
;	code/config/Config.c:26: }
	ret
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_SET_FREQ'
;------------------------------------------------------------
;freq          Allocated to registers 
;------------------------------------------------------------
;	code/config/Config.c:31: void CONF_SET_FREQ(uint16_t freq)
;	-----------------------------------------
;	 function CONF_SET_FREQ
;	-----------------------------------------
_CONF_SET_FREQ:
	mov	_sys_freq,dpl
	mov	(_sys_freq + 1),dph
;	code/config/Config.c:33: sys_freq = freq;
;	code/config/Config.c:34: }
	ret
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_CHANGE_SLEEP_MODE'
;------------------------------------------------------------
;	code/config/Config.c:39: void CONF_CHANGE_SLEEP_MODE()
;	-----------------------------------------
;	 function CONF_CHANGE_SLEEP_MODE
;	-----------------------------------------
_CONF_CHANGE_SLEEP_MODE:
;	code/config/Config.c:42: if (IapReadByte(addr_sleep_mode) & 0x01)
	mov	dptr,#0x0400
	lcall	_IapReadByte
	mov	a, dpl
	jnb	acc.0,00102$
;	code/config/Config.c:44: sys_sleep_mode = 0;
	mov	_sys_sleep_mode,#0x00
	sjmp	00103$
00102$:
;	code/config/Config.c:48: sys_sleep_mode = 1;
	mov	_sys_sleep_mode,#0x01
00103$:
;	code/config/Config.c:51: IapEraseSector(addr_sleep_mode);
	mov	dptr,#0x0400
	lcall	_IapEraseSector
;	code/config/Config.c:52: IapProgramByte(addr_sleep_mode, sys_sleep_mode);
	mov	_IapProgramByte_PARM_2,_sys_sleep_mode
	mov	dptr,#0x0400
;	code/config/Config.c:53: }
	ljmp	_IapProgramByte
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_RADIO_ERASE'
;------------------------------------------------------------
;	code/config/Config.c:58: void CONF_RADIO_ERASE()
;	-----------------------------------------
;	 function CONF_RADIO_ERASE
;	-----------------------------------------
_CONF_RADIO_ERASE:
;	code/config/Config.c:60: }
	ret
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_RADIO_PUT'
;------------------------------------------------------------
;freq          Allocated with name '_CONF_RADIO_PUT_PARM_2'
;index         Allocated to registers 
;------------------------------------------------------------
;	code/config/Config.c:65: void CONF_RADIO_PUT(uint8_t index, uint16_t freq)
;	-----------------------------------------
;	 function CONF_RADIO_PUT
;	-----------------------------------------
_CONF_RADIO_PUT:
;	code/config/Config.c:67: }
	ret
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_SET_INDEX_MAX'
;------------------------------------------------------------
;index         Allocated to registers 
;------------------------------------------------------------
;	code/config/Config.c:73: void CONF_SET_INDEX_MAX(uint8_t index)
;	-----------------------------------------
;	 function CONF_SET_INDEX_MAX
;	-----------------------------------------
_CONF_SET_INDEX_MAX:
;	code/config/Config.c:75: }
	ret
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_SYS_INIT'
;------------------------------------------------------------
;	code/config/Config.c:77: uint8_t CONF_SYS_INIT(void)
;	-----------------------------------------
;	 function CONF_SYS_INIT
;	-----------------------------------------
_CONF_SYS_INIT:
;	code/config/Config.c:90: return 0;
	mov	dpl, #0x00
;	code/config/Config.c:91: }
	ret
;------------------------------------------------------------
;Allocation info for local variables in function 'CONF_GET_RADIO_INDEX'
;------------------------------------------------------------
;index         Allocated to registers 
;------------------------------------------------------------
;	code/config/Config.c:96: uint16_t CONF_GET_RADIO_INDEX(uint8_t index)
;	-----------------------------------------
;	 function CONF_GET_RADIO_INDEX
;	-----------------------------------------
_CONF_GET_RADIO_INDEX:
;	code/config/Config.c:98: return 0;
	mov	dptr,#0x0000
;	code/config/Config.c:99: }
	ret
	.area CSEG    (CODE)
	.area CONST   (CODE)
	.area XINIT   (CODE)
	.area CABS    (ABS,CODE)
